clients/src/main/java/org/apache/kafka/clients/producer/KafkaProducer.java

```diff
@@ -468,7 +468,7 @@ private static int parseAcks(String acksString) {
             }
 
             int partition = partition(record, serializedKey, serializedValue, cluster);
-            int serializedSize = Records.LOG_OVERHEAD + Record.recordSize(serializedKey, serializedValue);
+            int serializedSize = EosLogEntry.entrySizeUpperBound(serializedKey, serializedValue);
```

Comment:
This is temporary. In a follow-up, we need to handle both magic versions in order to support backwards compatibility.